ONSdigital/eq-questionnaire-runner
# Remove save on signout code

### What is the context of this PR?
Removes redundant save on signout code and tests. This hasn't been used since the signout button was changed to a link in https://github.com/ONSdigital/eq-survey-runner/pull/2277.

As part of this work a new `signed-out` endpoint has been added to differentiate it from `sign-out`. `sign-out` does the work of signing out and redirecting to either a provided log out url or the `signed-out` page.

The sign out button logic has been simplified to remove the check for an `account_service_log_out_url`. This was to hide the sign out button on the `thank-you` page. Instead there is now an explicit property `hide_signout_button` and cookie state modification on the GET of the `thank-you` page has been removed.

### How to review 
- Check that all code related to save on signout has been removed
- Check that tests that have been removed should have been

### Checklist
